src/Model/List/SS_List.php

```diff
@@ -15,83 +16,243 @@
  */
 interface SS_List extends ArrayAccess, Countable, IteratorAggregate
 {
-
     /**
      * Returns all the items in the list in an array.
      *
      * @return array<T>
      */
-    public function toArray();
+    public function toArray(): array;
 
     /**
      * Returns the contents of the list as an array of maps.
-     *
-     * @return array
      */
-    public function toNestedArray();
+    public function toNestedArray(): array;
 
     /**
      * Adds an item to the list, making no guarantees about where it will
      * appear.
-     *
-     * @param mixed $item
      */
-    public function add($item);
+    public function add(mixed $item): void;
 
     /**
      * Removes an item from the list.
-     *
-     * @param mixed $item
      */
-    public function remove($item);
+    public function remove(mixed $item);
+
+    /**
+     * Return the number of items in this list
+     */
+    public function count(): int;
 
     /**
      * Returns the first item in the list.
      *
      * @return T|null
      */
-    public function first();
+    public function first(): mixed;
 
     /**
      * Returns the last item in the list.
      *
      * @return T|null
      */
-    public function last();
+    public function last(): mixed;
 
     /**
      * Returns a map of a key field to a value field of all the items in the
      * list.
-     *
-     * @param  string $keyfield
-     * @param  string $titlefield
-     * @return Map
      */
-    public function map($keyfield = 'ID', $titlefield = 'Title');
+    public function map(string $keyfield = 'ID', string $titlefield = 'Title'): Map;
 
     /**
      * Returns the first item in the list where the key field is equal to the
      * value.
      *
-     * @param  string $key
-     * @param  mixed $value
      * @return T|null
      */
-    public function find($key, $value);
+    public function find(string $key, mixed $value): mixed;
 
     /**
      * Returns an array of a single field value for all items in the list.
-     *
-     * @param  string $colName
-     * @return array
      */
-    public function column($colName = "ID");
+    public function column(string $colName = "ID"): array;
+
+    /**
+     * Returns a unique array of a single field value for all items in the list.
+     */
+    public function columnUnique(string $colName = 'ID'): array;
 
     /**
      * Walks the list using the specified callback
      *
-     * @param callable $callback
-     * @return static<T>
+     * @return SS_List<T>
+     */
+    public function each(callable $callback): SS_List;
+
+    /**
+     * Returns TRUE if the list can be filtered by a given field expression.
+     */
+    public function canFilterBy(string $by): bool;
+
+    /**
+     * Returns true if this list has items
+     */
+    public function exists(): bool;
```

Comment:
If we're including this, then remove the conditional check on `ListDecorator` for it. Though I don't think we need to include it personally.

Reply:
Removed the check in ListDecorator